modules/apps/knowledge-base/knowledge-base-web/src/main/java/com/liferay/knowledge/base/web/internal/display/context/KBArticleItemSelectorViewDisplayContext.java

```diff
@@ -0,0 +1,569 @@
+				kbFolderIds, articleAndFolderSearchContainer.getStart(),
+				articleAndFolderSearchContainer.getEnd(), sort);
+
+			Hits hits = indexer.search(searchContext);
+
+			List<Object> results = new ArrayList<>();
+
+			Document[] documents = hits.getDocs();
+
+			for (Document document : documents) {
+				String className = document.get(Field.ENTRY_CLASS_NAME);
+				long classPK = GetterUtil.getLong(
+					document.get(Field.ENTRY_CLASS_PK));
+
+				if (className.equals(KBArticle.class.getName())) {
+					results.add(
+						KBArticleLocalServiceUtil.fetchLatestKBArticle(
+							classPK, WorkflowConstants.STATUS_APPROVED));
+				}
+				else if (className.equals(KBFolder.class.getName())) {
+					results.add(KBFolderLocalServiceUtil.getKBFolder(classPK));
+				}
+			}
+
+			articleAndFolderSearchContainer.setResultsAndTotal(
+				() -> results, hits.getLength());
+		}
+		else {
+			articleAndFolderSearchContainer.setResultsAndTotal(
+				() -> {
+					OrderByComparator<Object> folderOrderByComparator = null;
+
+					boolean orderByAsc = false;
+
+					if (Objects.equals(_getOrderByType(), "asc")) {
+						orderByAsc = true;
+					}
+
+					if (Objects.equals(_getOrderByCol(), "relevance")) {
+						folderOrderByComparator =
+							new KBObjectsPriorityComparator<>(orderByAsc);
+					}
+					else if (Objects.equals(
+								_getOrderByCol(), "modified-date")) {
+
+						folderOrderByComparator =
+							new KBObjectsModifiedDateComparator<>(orderByAsc);
+					}
+					else if (Objects.equals(_getOrderByCol(), "title")) {
+						folderOrderByComparator =
+							new KBObjectsTitleComparator<>(orderByAsc);
+					}
+
+					return KBFolderServiceUtil.getKBFoldersAndKBArticles(
+						_getGroupId(), _getKBFolderId(),
+						_infoItemItemSelectorCriterion.getStatus(),
+						articleAndFolderSearchContainer.getStart(),
+						articleAndFolderSearchContainer.getEnd(),
+						folderOrderByComparator);
+				},
+				KBFolderServiceUtil.getKBFoldersAndKBArticlesCount(
+					_getGroupId(), _getKBFolderId(),
+					_infoItemItemSelectorCriterion.getStatus()));
+		}
+
+		_articleSearchContainer = articleAndFolderSearchContainer;
+
+		return _articleSearchContainer;
+	}
+
+	public boolean isSearch() {
+		if (_isEverywhereScopeFilter()) {
+			return true;
+		}
+
+		return _search;
+	}
+
+	public boolean isSearchEverywhere() {
+		if (_searchEverywhere != null) {
+			return _searchEverywhere;
+		}
+
+		if (Objects.equals(
+				ParamUtil.getString(_httpServletRequest, "scope"),
+				"everywhere")) {
+
+			_searchEverywhere = true;
+		}
+		else {
+			_searchEverywhere = false;
+		}
+
+		return _searchEverywhere;
+	}
+
+	protected SearchContext buildSearchContext(
+			List<Long> kbFolderIds, int start, int end, Sort sort)
+		throws PortalException {
+
+		SearchContext searchContext = new SearchContext();
+
+		searchContext.setAndSearch(false);
+		searchContext.setAttribute(
+			Field.CLASS_NAME_ID, KBArticleConstants.CLASS_NAME_ID_DEFAULT);
```

Comment:
Why is it necessary to set the class name id to 0? Isn't this equivalent to leaving it uninitialized?

Reply:
put to `0` it's because it can be folder or articles
But I do not know if it's needed. 
I will remove it and see it something fails :D